hierarchicalforecast/methods.py

```diff
@@ -822,7 +822,7 @@ def _get_PW_matrices(self,
                 corm = np.nan_to_num(corm, nan=0.0)
                 xs = np.divide(residuals, residual_std, 
                                out=np.zeros_like(residuals), where=residual_std!=0)
-
+                
                 xs = xs[~np.isnan(xs).any(axis=1), :]
                 v = (1 / (n * (n - 1))) * (crossprod(xs ** 2) - (1 / n) * (crossprod(xs) ** 2))
```

Comment:
```suggestion
                v = (n / ((n - 1) ** 3)) * (crossprod(xs ** 2) - (1 / n) * (crossprod(xs) ** 2))
```

I know this is not changed in the PR, but is it me or does this look wrong? ``np.ma.cov`` gives an unbiased estimate by default, and this would only be valid for a biased estimate using $`N`$ degrees of freedom.

Reply:
I was under the impression that this is a consolidation of factors (see below), but I also remember that when I checked a while ago the unbiasedness isn't completely correct in the original implementation in R (our original follows the original Hyndman implementation from R). 

In the original Schafer-Strimmer algorithm:
```
factor_emp_corr = (n_samples / (n_samples - 1)
factor_var_emp_cor = (n_samples / (n_samples - 1)**3
for i ....
    for j ....
       sum_var_emp_corr += 2 * factor_var_emp_cor * np.sum(np.square(w - w_mean))
       sum_sq_emp_corr += 2 * np.square(factor_emp_corr * w_mean)

shrinkage = sum_var_emp_corr / (sum_sq_emp_corr + epsilon)
```

Now, we can equivalently write (we can ignore the epsilon, that's there for stability purposes):
```
factor_shrinkage = 1 / (n_samples * (n_samples - 1))
for i ....
    for j ....
       sum_var_emp_corr += np.sum(np.square(w - w_mean))
       sum_sq_emp_corr += w_mean**2

shrinkage = (factor_shrinkage * sum_var_emp_corr) / (sum_sq_emp_corr + epsilon)
```

I think I'll leave it as it is for now.